lib/src/main/kotlin/WeatherSensorService/Public/ClosedRangeExtension.kt

```diff
@@ -0,0 +1,22 @@
+ //
+ // SpinDance
+ // Weather Sensor Service
+ //
+ // Copyright © 2022 SpinDance. All rights reserved.
+ //
+
+ import kotlin.random.Random
+ import kotlin.math.sin
+ import kotlin.math.abs
+ 
+ public fun ClosedRange<Double>.calculateSineWaveReading(cyclePercentage: Double): Double {
+    // Will be between -1 and 1 with the value determined based on the completion value of cyclePercentage
+    val coefficient = sin(2 * Math.PI * cyclePercentage)
+ 
+    // median of the amplitude
+    val median = abs(this.endInclusive - this.start) / 2
+ 
+    // Applies the coefficient and median amplitude to create a sin wave based around the range provided
+    return this.start + median + coefficient * median
```

Comment:
I hope I didn't mislead, I think most coding standard would still want the () around the multiplication, there's usually a rule about not relying upon the implied "order of operations", so as to be explicit about the order. I think before you had that set of () plus another... it was the other one I'd commented on.

Reply:
Pretty sure you don't need `this` here.